### PR TITLE
Add option to use the old legend style

### PR DESCRIPTION
**Pull request by elpaso**
Makes it possible to use the "old" legend style (separate items for each value).

The addition to the "Legend Settings" dialog is the top checkbox, when unchecked the other widget elements are disabled.

immagine

### CODE REVIEW

File: python/core/auto_generated/layertree/qgscolorramplegendnodesettings.sip.in [hunk 1]
```diff
@@ -197,6 +197,26 @@ Sets the ramp ``orientation`` (i.e. horizontal or vertical).
 .. seealso:: :py:func:`orientation`
 
 .. seealso:: :py:func:`setDirection`
+%End
+
+    bool useContinuousLegend() const;
+%Docstring
+Returns the ``True`` if the flag to use continuous legend is set.
```
Comment (nyalldawson) [inline python/core/auto_generated/layertree/qgscolorramplegendnodesettings.sip.in:204]:
```suggestion
Returns ``True`` if a continuous gradient legend will be used.
```

(And similar for the setter)

**Review by nyalldawson — APPROVED** [review]
Can you extend the existing tests to cover this new setting too?

**Comment by elpaso** [comment]
> Can you extend the existing tests to cover this new setting too?

@nyalldawson I had a look but couldn't find any test for the settings class `QgsColorRampLegendNodeSettings` where do you think new tests should be added?

The changes are basically restoring the old behavior that I assume is already tested for the other interpolation methods, there are no changes within the new `QgsColorRampLegendNode` class.

**Comment by nyalldawson** [comment]
@elpaso

>  I had a look but couldn't find any test for the settings class QgsColorRampLegendNodeSettings where do you think new tests should be added?

use the grep :mage_man: 
https://github.com/qgis/QGIS/blob/master/tests/src/python/test_qgscolorramplegendnode.py#L65

**Comment by tomas4** [comment]
I tested this on Linux/Debian with QGIS 3.18.1-Zürich Rev. 202f1bf7e5. It works as expected (at least by me).

**Pull request merged**